core/referral/snapshot_test.go

```diff
@@ -189,6 +188,21 @@ func TestTakingAndRestoringSnapshotSucceeds(t *testing.T) {
 	// Simulate restoration of the network parameter at the time of the snapshot
 	require.NoError(t, te2.engine.OnReferralProgramMaxPartyNotionalVolumeByQuantumPerEpochUpdate(ctx, maxVolumeParams))
 
+	// OnStateLoaded will recalculate referrer stats and send out all these events
+	te2.marketActivityTracker.EXPECT().NotionalTakerVolumeForParty(string(referrer1)).Return(num.UintFromUint64(100)).Times(1)
```

Comment:
why not `Times(2)` for referrers 1-4? Or just create a matcher type that matches all expected calls exactly once?